cli/flox/src/commands/publish.rs

```diff
@@ -140,3 +139,153 @@ fn check_package(lockfile: &Lockfile, package: &str) -> Result<bool> {
 
     Ok(true)
 }
+
+/// Merge cache values from config and args, with args taking precedence.
+/// Values must be mutually present or absent.
+fn merge_cache_options(
+    config: Option<PublishConfig>,
+    args: Option<CacheArgs>,
+) -> Result<Option<NixCopyCache>> {
+    match (
+        args.as_ref()
```

Comment:
Nit: a match expression this large makes it a little tough to read the structure of what's being matched. Factoring it out to a separate expression so that you can `match (foo, bar)` is my preference.

Reply:
I've pulled them out to separate bindings beforehand.